src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -211,19 +214,48 @@ private SwiftCompileStep makeCompileStep(SourcePathResolver resolver) {
             .anyMatch(SwiftDescriptions.SWIFT_MAIN_FILENAME::equalsIgnoreCase);
 
     compilerCommand.add(
-        "-c",
         enableObjcInterop ? "-enable-objc-interop" : "",
         hasMainEntry ? "" : "-parse-as-library",
         "-serialize-debugging-options",
         "-module-name",
-        moduleName,
-        "-emit-module",
-        "-emit-module-path",
-        modulePath.toString(),
-        "-emit-objc-header-path",
-        headerPath.toString(),
-        "-o",
-        objectFilePath.toString());
+        moduleName);
+
+    if (swiftBuckConfig.shouldSplitSwiftModuleGeneration()) {
+      compilerCommand.add(
+          "-emit-module",
+          "-emit-module-path",
+          modulePath.toString(),
+          "-emit-objc-header",
+          "-emit-objc-header-path",
+          headerPath.toString());
+
+      if (moduleOnly) {
+        compilerCommand.add("-emit-bc");
+        compilerCommand.add("-num-threads", "1");
+        this.srcs
+            .stream()
+            .map(
+                input -> getBitcodePath(resolver.getAbsolutePath(input).getFileName().toString())
+            )
+            .forEach(
+                x -> {
+                  compilerCommand.add("-o");
+                  compilerCommand.add(x.toString());
```

Comment:
@robbertvanginkel this means we will have collisions if there are multiple files with the same name in a target. Better to store them with BUCK relative paths?

Reply:
Not an issue, swift requires unique filenames when compiling a module:
```
$ swiftc temp/test.swift test.swift
<unknown>:0: error: filename "test.swift" used twice: 'temp/test.swift' and 'test.swift'
<unknown>:0: note: filenames are used to distinguish private declarations with the same name
```